src/trousse/feature_fix.py

```diff
@@ -241,27 +239,19 @@ def _ordinal_encode_column(df, column, drop_old_column: bool = False):
     -------
 
     """
-    df_new = df.copy()
-    series = df_new[column].values
-    # Adding a new dimension for use with OrdinalEncoder (1D array to column vector shape=(5,1))
-    series = series[..., np.newaxis]
-    encoder = OrdinalEncoder()
-    encoder_fitted = encoder.fit(series)  # Encoder object (for reverse transformation)
-    series_enc = encoder_fitted.transform(series)
-    new_column = f"{column}{ENCODED_COLUMN_SUFFIX}"
-    # Convert the encoded columns to Integer type (this pandas Dtype is for handling NaN values)
-    df_new[new_column] = series_enc
-    df_new[new_column] = df_new[new_column].astype(pd.Int16Dtype())
 
-    if drop_old_column:
-        df_new = df_new.drop(column, axis=1)
-    return df_new, encoder_fitted, [new_column]
+    dataset = Dataset(df_object=df)
+    derived_column = f"{column}_enc"
+    ordinal_encoder = OrdinalEncoder(columns=[column], derived_columns=[derived_column])
+
+    encoded_dataset = ordinal_encoder(dataset)
+    return encoded_dataset.data, ordinal_encoder.encoder, [derived_column]
 
 
 def encode_single_categorical_column(
     dataset: Dataset,
     col_name: str,
-    encoding: EncodingFunctions = EncodingFunctions.ORDINAL,
+    encoding: Any = "EncodingFunctions.ORDINAL",
```

Comment:
Are we sure that we want to remove the Enum usage? It looked useful for me and also Valerio seemed to support that.
By the way I think this is wrong and we should replace `EncodingFunctions.ORDINAL` with `ORDINAL`